packages/core/threads/thingsboard_thread.py

```diff
@@ -0,0 +1,104 @@
+        config = types.Config.load()
+        assert config.thingsboard is not None
+
+        # initialize MQTT client
+        client = TBDeviceMqttClient(
+            config.thingsboard.host, username=config.thingsboard.access_token
+        )
+
+        # connect to MQTT broker
+        if config.thingsboard.ca_cert:
+            client.connect(tls=True, ca_certs=config.thingsboard.ca_cert)
+        else:
+            client.connect()
+
+        logger.info("Succesfully connected to Thingsboard Broker.")
+
+        while True:
+            # Read latest config
+            config = types.Config.load()
+            if not ThingsBoardThread.should_be_running(config):
+                logger.info("ThingsboardThread shall not be running anymore.")
+                client.disconnect()
+                return
+
+            # publish if client is connected
+            if not client.is_connected():
+                logger.warning(
+                    "Client is currently not connected. Waiting for reconnect."
+                )
+            else:
+                # read latest state file
+                current_state = interfaces.StateInterface.load_state()
+                state: Dict[str, Optional[Union[str, bool, int, float]]] = {
+                    "last_updated": str(current_state.last_updated),
+                    "helios_indicates_good_conditions": current_state.helios_indicates_good_conditions,
+                    "measurements_should_be_running": current_state.measurements_should_be_running,
+                    "memory_usage": current_state.operating_system_state.memory_usage,
+                    "filled_disk_space_fraction": current_state.operating_system_state.filled_disk_space_fraction,
+                    "fan_speed": current_state.plc_state.actors.fan_speed,
+                    "current_angle": current_state.plc_state.actors.current_angle,
+                    "auto_temp_mode": current_state.plc_state.control.auto_temp_mode,
+                    "manual_control": current_state.plc_state.control.manual_control,
+                    "manual_temp_mode": current_state.plc_state.control.manual_temp_mode,
+                    "sync_to_tracker": current_state.plc_state.control.sync_to_tracker,
+                    "humidity": current_state.plc_state.sensors.humidity,
+                    "temperature": current_state.plc_state.sensors.temperature,
+                    "cover_closed": current_state.plc_state.state.cover_closed,
+                    "motor_failed": current_state.plc_state.state.motor_failed,
+                    "rain": current_state.plc_state.state.rain,
+                    "reset_needed": current_state.plc_state.state.reset_needed,
+                    "ups_alert": current_state.plc_state.state.ups_alert,
+                    "heater": current_state.plc_state.power.heater,
+                    "spectrometer": current_state.plc_state.power.spectrometer,
+                }
```

Comment:
The dict keys should be a little more descriptive. For example `spectrometer_power` instead of `spectrometer`.